Xplat/src/main/java/vazkii/botania/common/entity/ManaSparkEntity.java

```diff
@@ -365,12 +416,24 @@ public SparkUpgradeType getUpgrade() {
 	@Override
 	public void setUpgrade(SparkUpgradeType upgrade) {
 		entityData.set(UPGRADE, upgrade.ordinal());
+		updateTransfers();
+		notifyOthers(getNetwork());
+	}
+
+	@Override
+	public void setNetwork(DyeColor color) {
+		// The previous network needs to filter this spark out
+		var previousNetwork = getNetwork();
+		super.setNetwork(color);
+		updateTransfers();
+		notifyOthers(color);
+		notifyOthers(previousNetwork);
 	}
 
 	@Override
 	public boolean areIncomingTransfersDone() {
 		if (getAttachedManaReceiver() instanceof ManaPool) {
-			return removeTransferants > 0;
```

Comment:
tbh I'm not really sure what this old code did lol, do you know and can you explain why it should be removed?

Reply:
RemoveTransferants was the old reset logic when stuff like the upgrade or the network changed. This is now handled differently. And the only reason why it might be removed is if it is full, which is checked seperately